ui/cypress/e2e/sampleControls.cy.js

```diff
@@ -1,42 +1,43 @@
-/* global cy, it, describe, beforeEach, Cypress */
+/* global cy, it, describe, beforeEach */
 
 describe('3-click centring', () => {
   beforeEach(() => {
-    cy.login();
-    cy.findByRole('heading', { name: 'MXCuBE-Web (OSC)' }).should('be.visible');
-    cy.takeControl();
+    cy.loginWithControl();
   });
 
-  it('3-click centring should not work without sample', () => {
-    cy.clearSamples();
+  it('displays error when no sample is mounted', () => {
+    cy.clearSamples(); // another test may have mounted a sample
+    cy.findByRole('link', { name: /Data collection/u, hidden: true }).click();
+
     cy.findByRole('button', { name: '3-click centring' }).click();
     cy.findByRole('alert', 'Error: There is no sample mounted').should(
       'be.visible',
     );
   });
 
-  it.skip('Each click is rotating the sample by 90 degrees', () => {
+  it('rotates sample by 90 degrees on click', () => {
     cy.mountSample();
     cy.findByRole('button', { name: 'Sample: test - test' }).should(
       'be.visible',
     );
+
     cy.findByRole('button', { name: '3-click centring' }).click();
+    cy.findByText(/Clicks left: 3/u, { hidden: true }).should('exist');
+
     cy.get('.form-control[name="diffractometer.phi"]')
       .invoke('val')
-      .then((value) => {
-        let omegaValue = Number.parseFloat(value);
-        Cypress._.times(2, () => {
```

Comment:
Note that I also simplify this test by performing only one rotation.

Reply:
It might not be a "3"-click centring test then, but for the sake of performance increase, I'm happy to reduce the test to a single rotation.

Reply:
I guess a more complete test would be to perform the 3-click centering and check that the point gets added. But  that could be a separate test I guess.

Reply:
Yeah exactly, but without WebSockets, it's too complicated a feature to test for now, in my opinion.